lib/api/core/models/repositories/repository.js

```diff
@@ -1,344 +1,421 @@
+    }
   }
-  this.cacheKeyPrefix = 'repos/' + this.index + '/' + this.collection + '/';
-};
 
-/**
- *
- * @param {string} id
- * @returns {Promise} resolves on a new ObjectConstructor()
- */
-Repository.prototype.loadOneFromDatabase = function repositoryLoadOneFromDatabase (id) {
-  var result;
-
-  return this.databaseEngine.get(this.collection, id)
-    .then(response => {
-      if (response._id) {
-        result = new this.ObjectConstructor();
-        if (response._source) {
-          return _.assignIn(result, response._source, {_id: response._id});
-        }
-        return _.assignIn(result, response);
-      }
-      return null;
-    })
-    .catch(error => {
-      if (error.status === 404) {
-        // no content found, we return null without failing
-        return Promise.resolve(null);
-      }
+  /**
+   * @param {string} id
+   * @param {string} [collection]
+   * @returns {Promise} resolves on a new ObjectConstructor()
+   */
+  loadOneFromDatabase (id, collection = this.collection) {
+    let result;
 
-      return Promise.reject(error);
-    });
-};
+    if (collection === null) {
+      return Promise.reject(new InternalError('The collection must be specified.'));
+    }
 
-/**
- *
- * @param _ids
- * @returns {*|promise}
- */
-Repository.prototype.loadMultiFromDatabase = function repositoryLoadMultiFromDatabase (_ids) {
-  var ids = [];
+    return this.databaseEngine.get(collection, id)
+      .then(response => {
+        if (response._id) {
+          result = new this.ObjectConstructor();
+          if (response._source) {
+            return _.assignIn(result, response._source, {_id: response._id});
+          }
+          return _.assignIn(result, response);
+        }
+        return null;
+      })
+      .catch(error => {
+        if (error.status === 404) {
+          // no content found, we return null without failing
+          return Promise.resolve(null);
+        }
 
-  if (!_.isArray(_ids)) {
-    return Promise.reject(new InternalError('Bad argument: ' + _ids.toString() + ' is not an array.'));
+        return Promise.reject(error);
+      });
   }
-  _ids.forEach(element => {
-    if (!_.isObject(element)) {
-      ids.push(element);
-    }
-    else {
-      ids.push(element._id);
+
+  /**
+   *
+   * @param {string[]|object[]} _ids
+   * @param {string} [collection]
+   * @returns {Promise<object>}
+   */
+  loadMultiFromDatabase (_ids, collection = this.collection) {
+    const ids = [];
+
+    if (collection === null) {
+      return Promise.reject(new InternalError('The collection must be specified.'));
     }
-  });
 
-  return this.databaseEngine.mget(this.collection, ids)
-    .then(response => {
-      if (!response.hits || response.hits.length === 0) {
-        return Promise.resolve([]);
+    if (!_.isArray(_ids)) {
+      return Promise.reject(new InternalError('Bad argument: ' + _ids.toString() + ' is not an array.'));
```

Comment:
It can not happen due to a bad request, it might come from an bad plugin implementation, will use this one instead.